robots/pkg/ci-usage-exporter/metrics/metrics_test.go

```diff
@@ -0,0 +1,679 @@
+									Name:    "test-job2",
+									Cluster: "test-cluster",
+									Spec: &v1.PodSpec{
+										Containers: []v1.Container{
+											{
+												Resources: v1.ResourceRequirements{
+													Requests: v1.ResourceList{
+														v1.ResourceMemory: quantity2,
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			[]metricExpectations{
+				{
+					name:  "kubevirt_ci_job_memory_bytes",
+					value: quantityValue1,
+					labels: map[string]string{
+						"job_name":    "test-job1",
+						"type":        "periodic",
+						"job_cluster": "test-cluster",
+					},
+				},
+				{
+					name:  "kubevirt_ci_job_memory_bytes",
+					value: quantityValue2,
+					labels: map[string]string{
+						"job_name":    "test-job2",
+						"type":        "periodic",
+						"job_cluster": "test-cluster",
+					},
+				},
+			},
+		),
+		Entry("presubmit, postsubmit, and periodic",
+			metrics.Config{
+				ProwConfig: &prowConfig.Config{
+					JobConfig: prowConfig.JobConfig{
+						PresubmitsStatic: map[string][]prowConfig.Presubmit{
+							"test-org1/test-repo1": {
+								{
+									JobBase: prowConfig.JobBase{
+										Name:    "test-job1",
+										Cluster: "test-cluster1",
+										Spec: &v1.PodSpec{
+											Containers: []v1.Container{
+												{
+													Resources: v1.ResourceRequirements{
+														Requests: v1.ResourceList{
+															v1.ResourceMemory: quantity1,
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+						PostsubmitsStatic: map[string][]prowConfig.Postsubmit{
+							"test-org2/test-repo2": {
+								{
+									JobBase: prowConfig.JobBase{
+										Name:    "test-job2",
+										Cluster: "test-cluster2",
+										Spec: &v1.PodSpec{
+											Containers: []v1.Container{
+												{
+													Resources: v1.ResourceRequirements{
+														Requests: v1.ResourceList{
+															v1.ResourceMemory: quantity2,
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+						Periodics: []prowConfig.Periodic{
+							{
+								JobBase: prowConfig.JobBase{
+									Name:    "test-job3",
+									Cluster: "test-cluster3",
+									Spec: &v1.PodSpec{
+										Containers: []v1.Container{
+											{
+												Resources: v1.ResourceRequirements{
+													Requests: v1.ResourceList{
+														v1.ResourceMemory: quantity3,
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			[]metricExpectations{
+				{
+					name:  "kubevirt_ci_job_memory_bytes",
+					value: quantityValue1,
+					labels: map[string]string{
+						"job_name":    "test-job1",
+						"org":         "test-org1",
+						"repo":        "test-repo1",
+						"type":        "postsubmit",
```

Comment:
Shouldn't this be `presubmit`?

Reply:
Absolutely, great catch! The regexps were matching too much, given that the labels always appear sorted alphabetically by name I've replaced regexp matching with simpler and more precise `strings.Contains`, PTAL.